dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MetadataDumper.java

```diff
@@ -302,15 +304,44 @@ public <T> T runChildTask(Task<T> task) throws MetadataDumperUsageException {
 
             if (requiredTasksNotSucceeded > 0) {
                 System.out.println(
-                        "* ERROR: " + requiredTasksNotSucceeded + " required task[s] failed.\n"
-                        + "* Output, including debugging information, has been saved to " + outputFile + "\n"
+                    "* ERROR: " + requiredTasksNotSucceeded + " required task[s] failed.\n"
+                        + "* Output, including debugging information, has been saved to "
+                        + outputFile + "\n"
                         + "**********************************************************"
                 );
-                if (exitOnError)
+                if (exitOnError) {
                     System.exit(1);
+                }
             }
 
+            logStatusSummary(state);
+            System.out.println("**********************************************************");
+        }
+    }
+
+    private void logStatusSummary(TaskSetState.Impl state) {
+        List<TaskState> taskStates = state.getTaskResultMap().values().stream()
+            .map(TaskResult::getState).collect(Collectors.toList());
+
+        Function<TaskState, Long> countState = desiredState -> taskStates.stream()
```

Comment:
Consider keeping a map of task state counts instead of list and a function:

```
Map<String, Long> taskStateCountMap = list.stream().map(TaskResult::getState)
        .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
```